tests/removal.bats

```diff
@@ -0,0 +1 @@
+test_helper/bats/removal.bats
```

Comment:
I wonder if we should rename the new test files and use a more generic term. For example lifecycle?

We could then at some point in the future extend the testing to cover the expected handling to `snap stop`, `snap refresh` in addition to `remove`.

WDYT?